tekton/src/tasks/gitops/gitops-mas-initiator.yml.j2

```diff
@@ -185,6 +190,20 @@ spec:
         # Update the channel in the mas-instance-params.yaml. Assume fvtsaas here
         yq --arg MAS_CHANNEL $MAS_CHANNEL -iY '.mas_instance.channel= $MAS_CHANNEL' gitops/$ACCOUNT/$REGION/$CLUSTER_NAME/fvtsaas/mas-instance-params.yaml
 
+        # Set dns.provider as empty in the mas-instance-params.yaml if deployment with cis is not enabled for that deployment day.  Assume fvtsaas here
+        export DNS_PROVIDER=$(yq -sr '.[0].dns.provider' gitops/$ACCOUNT/$REGION/$CLUSTER_NAME/fvtsaas/mas-instance-params.yaml)
+        export CURRENT_DATE=$(date +"%u")
+        export EMPTY_DNS_PROVIDER=""
+        echo "gitops-mas-initiator: DNS_PROVIDER=${DNS_PROVIDER} CURRENT_DATE=${CURRENT_DATE} DEPLOYMENT_DAY_WITH_CIS=$DEPLOYMENT_DAY_WITH_CIS"
+
+        if [ "$DNS_PROVIDER" == "cis" ]; then
+          if [[ -n $DEPLOYMENT_DAY_WITH_CIS && "$DEPLOYMENT_DAY_WITH_CIS" != "CURRENT_DATE" ]]; then
+            yq --arg EMPTY_DNS_PROVIDER "${EMPTY_DNS_PROVIDER}" -iY '(. | select(has("dns")?).dns.provider)= $EMPTY_DNS_PROVIDER' gitops/$ACCOUNT/$REGION/$CLUSTER_NAME/fvtsaas/mas-instance-params.yaml
+            yq -iY 'del(.dns.cis)' gitops/$ACCOUNT/$REGION/$CLUSTER_NAME/fvtsaas/mas-instance-params.yaml
```

Comment:
I think to conform to the schema we need to remove the whole `dns` section from the params file rather than leave `dns.provider: ""` as that technically breacks the schema.